app/forms/choose_role_form.rb

```diff
@@ -47,21 +44,44 @@ def redirect_path(helpers:)
   end
 
   def only_one_role
-    return false unless user.user_roles.count == 1
+    return false unless sanitized_user_roles.count == 1
 
-    self.role = user.user_roles.first
+    self.role = sanitized_user_roles.first
     true
   end
 
   def has_no_role
-    return false unless user.user_roles.count.zero?
+    return false unless sanitized_user_roles.empty?
 
     self.role = "no_role"
     true
   end
 
 private
 
+  def sanitized_user_roles
+    roles = user_roles.reject { |role| role.in? %w[induction_coordinator induction_coordinator_and_mentor teacher] }
+
+    if sit_role?
+      roles << "induction_coordinator"
+    elsif teacher_role?
+      roles << "teacher"
+    end
+    roles
+  end
```

Comment:
It might be better to move this to `User.user_roles`
And since we no longer have `induction_coordinator_and_mentor` role, should we look to remove the `induction_coordinator_mentor_path(user)` section? no one would be able to visit those pages.

Reply:
So I think those are probably legit roles just that we don't want to treat them separately here.  I don't know if anyone else uses `user_roles` for anything?  Really the `user_roles` is a duplicate/adaption of `user_description` could be completely handled here instead?  What do you think?

Reply:
Good spot about the paths will look at those

Reply:
The method `User.user_roles` needs to be updated in how we calculated the roles for a user.
It might be better to split up `mentor` into its own role. That way we can still assign it to a user that requires it.

`user_description` is not related to `user_roles`, we don't really use it anywhere I don't think other than `AnalyticsDataLayer`.

Reply:
The roles are used to allow users to visit different sections of the app and are used by Pundit for authorisation.

We assign the roles by creating the profile associations, eg: `AdminProfile`, `FinanceProfile`, `InductionCoordinatorProfile` etc